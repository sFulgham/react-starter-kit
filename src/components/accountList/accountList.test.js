import React from 'react';
import AccountList from './';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const testProps = {};
Enzyme.configure({adapter: new Adapter()});

beforeEach(() => {
  testProps.accounts = {
    '1': {id: '1', business: 'test', date: new Date().toString(), website: 'test.com', name: 'test'},
    '2': {id: '2', business: 'test', date: new Date().toString(), website: 'test.com', name: 'test'},
    '3': {id: '3', business: 'test', date: new Date().toString(), website: 'test.com', name: 'test'}
  };
});

describe('AccountList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <AccountList {...testProps} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('AccountContainer').length).toBe(3);
    console.log(wrapper); //eslint-disable-line no-console
  });
});