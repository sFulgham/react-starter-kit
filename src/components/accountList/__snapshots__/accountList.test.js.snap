// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountList
    accounts={
      Object {
        "1": Object {
          "business": "test",
          "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
          "id": "1",
          "name": "test",
          "website": "test.com",
        },
        "2": Object {
          "business": "test",
          "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
          "id": "2",
          "name": "test",
          "website": "test.com",
        },
        "3": Object {
          "business": "test",
          "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
          "id": "3",
          "name": "test",
          "website": "test.com",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AccountContainer
          account={
            Object {
              "business": "test",
              "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
              "id": "1",
              "name": "test",
              "website": "test.com",
            }
          }
          store={undefined}
        />,
        <AccountContainer
          account={
            Object {
              "business": "test",
              "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
              "id": "2",
              "name": "test",
              "website": "test.com",
            }
          }
          store={undefined}
        />,
        <AccountContainer
          account={
            Object {
              "business": "test",
              "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
              "id": "3",
              "name": "test",
              "website": "test.com",
            }
          }
          store={undefined}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "account": Object {
            "business": "test",
            "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
            "id": "1",
            "name": "test",
            "website": "test.com",
          },
          "store": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "account": Object {
            "business": "test",
            "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
            "id": "2",
            "name": "test",
            "website": "test.com",
          },
          "store": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "account": Object {
            "business": "test",
            "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
            "id": "3",
            "name": "test",
            "website": "test.com",
          },
          "store": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AccountContainer
            account={
              Object {
                "business": "test",
                "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
                "id": "1",
                "name": "test",
                "website": "test.com",
              }
            }
            store={undefined}
          />,
          <AccountContainer
            account={
              Object {
                "business": "test",
                "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
                "id": "2",
                "name": "test",
                "website": "test.com",
              }
            }
            store={undefined}
          />,
          <AccountContainer
            account={
              Object {
                "business": "test",
                "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
                "id": "3",
                "name": "test",
                "website": "test.com",
              }
            }
            store={undefined}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "account": Object {
              "business": "test",
              "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
              "id": "1",
              "name": "test",
              "website": "test.com",
            },
            "store": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "account": Object {
              "business": "test",
              "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
              "id": "2",
              "name": "test",
              "website": "test.com",
            },
            "store": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "account": Object {
              "business": "test",
              "date": "Sat Dec 30 2017 21:01:06 GMT-0500 (EST)",
              "id": "3",
              "name": "test",
              "website": "test.com",
            },
            "store": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
